### PR TITLE
Launch the classifier training

### LINKED ISSUE #5: Add documentation

### LINKED ISSUE #109: [FEATURE REQUEST] Let GCN classifier accept a list of hidden layers
**Please describe the GRACE feature you'd like to add**

Change the hard-coded 3-layered GCN convolutions with static hidden channels into a dynamic list

### LINKED ISSUE #110: [FEATURE REQUEST] Logging report to GCN training
**Please describe the GRACE feature you'd like to add**

At train time, the data pre-processing step, esp. patch cropping at nodes + feature extraction, take a lot of time (even with as little as 10 images) without informing the user that the training run is not stuck. Perhaps logger with description of the training progress would be helpful. Possibly, a progress bar with 1/10 images etc.

### PR DESCRIPTION
**Pull request by KristinaUlicna**
PR to enable GCN classifier training.

- [x] [GCN] Rename model runs time stamp to save into folder chronologically
- [x] [GCN] Rename `val_fraction` to `train_fraction` argument, which is what it is
- [x] [GCN] Rename `ignore_fraction` to `keep_patch_fraction` argument, which is what it is
- [x] [GCN] Extend hidden layers argument to list, e.g. `[8, 16, 32, 64]` - resolves #109 
- [x] [GCN] Write config hyperparameters documentation - resolves #5 
- [x] [GCN] Add an example config file to illustrate how to write one.
- [x] [GCN] Progress bar for image loading + feature extraction (tqdm) - resolves #110

### CODE REVIEW

**Review by mooniean — APPROVED** [review]
forgot to press approve

File: grace/models/classifier.py [hunk 1]
```diff
@@ -31,32 +31,46 @@ class GCN(torch.nn.Module):
     def __init__(
         self,
         input_channels: int,
-        hidden_channels: int,
+        hidden_channels: list[int],
```
Comment (chris-soelistyo) [inline grace/models/classifier.py:34]: Use `List` from `typing` instead of `list`

File: grace/models/classifier.py [hunk 2]
```diff
@@ -68,3 +82,17 @@ def forward(
         edge_x = self.edge_classifier(edge_features)
 
         return node_x, edge_x
+
+    def predict(
```
Comment (chris-soelistyo) [inline grace/models/classifier.py:86]: If the code doesn't do anything, maybe leave it for another PR?

File: grace/models/datasets.py [hunk 3]
```diff
@@ -8,40 +8,54 @@
 from grace.base import GraphAttrs, Annotation
 
 
-def dataset_from_graph(
-    graph: nx.Graph, *, n_hop: int = 1
+def dataset_from_subgraphs(
```
Comment (chris-soelistyo) [inline grace/models/datasets.py:11]: Can you remind me why the distinction was required of separate functions for subgraph and whole graph?

**Review by chris-soelistyo — COMMENTED** [review]
Looks good! Just have a few comments

**Comment by quantumjot** [comment]
@chris-soelistyo - I would go ahead and merge this if you're happy with it
